### ISSUE TITLE
Agree and adopt a Code of Conduct for Iris

### ISSUE THREAD

**Issue by bjlittle** [SciTools/iris#5183]
## 📰 Custom Issue

I think that we should have a code of conduct for the `iris` community, which is highlighted as a good practice in the GH repo community standards
![image](https://user-images.githubusercontent.com/2051656/222120452-da30580d-9172-4630-b8fb-2c54c3cac57e.png)

There are various "off-the-shelf" code of conducts that could be adopted, as suggested by GH i.e.,

- the [Contributor Covenant](https://www.contributor-covenant.org/)
- the [Citizen Code of Conduct](https://github.com/stumpsyn/policies/blob/master/citizen_code_of_conduct.md)

Alternatively, some packages choose to adopt the same or adapt (i.e., draw inspiration from) code of conducts from other leading packages, such as:

- the [NumPy Code of Conduct](https://numpy.org/code-of-conduct/)
- the [SciPy Code of Conduct](https://docs.scipy.org/doc/scipy/dev/conduct/code_of_conduct.html)

I'm sure there are more to choose from if we seriously looked... but in general I'd like to believe that having a code of conduct is way better than having no code of conduct.

Note that, `xarray` choose to follow the `Contributor Covenant`, as do `matplotlib`.

Thoughts?

**Comment by bjlittle** [comment]
To start the ball rolling... 

1. I'd personally vote for adopting the [Contributor Covenant](https://www.contributor-covenant.org/version/2/1/code_of_conduct/) v2.1 (latest)
2. Raise a PR on the [Contributor Covenant](https://github.com/EthicalSource/contributor_covenant) repo [Registering Your Community as an Adopter](https://github.com/EthicalSource/contributor_covenant/blob/release/README.md#registering-your-community-as-an-adopter).

**Comment by trexfeathers** [comment]
I'll need to do more reading later, but I expect this is in the same space:

https://fair-software.eu/

[![fair-software.eu](https://img.shields.io/badge/fair--software.eu-%E2%97%8F%20%20%E2%97%8F%20%20%E2%97%8F%20%20%E2%97%8F%20%20%E2%97%8B-yellow)](https://fair-software.eu)

I've seen a few of these badges around.

**Comment by bjlittle** [comment]
> I need to do more reading later, but I expect this is in the same space:
> 
> https://fair-software.eu/
> 
> [![fair-software.eu](https://camo.githubusercontent.com/9a74b21b0c4c076977018603e207add4af89be37d4b549c3f1061ed718771003/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f666169722d2d736f6674776172652e65752d2545322539372538462532302532302545322539372538462532302532302545322539372538462532302532302545322539372538462532302532302545322539372538422d79656c6c6f77)](https://fair-software.eu)
> 
> I've seen a few of these badges around.

That's from the [eScience Center](https://www.esciencecenter.nl/) where @bouweandela works

**Comment by rcomer** [comment]
On one occasion so far I have had reason to point to Matplotlib's Code of Conduct, and was kinda glad it was there.  It may never come up, but the effort to add one seems minimal, so 👍 from me.

**Comment by bouweandela** [comment]
> That's from the [eScience Center](https://www.esciencecenter.nl/) where @bouweandela works (if I'm not mistaken)

Yes, some of my colleagues are very active in building and promoting things to make research software more FAIR. If you have any questions about this, I'd be happy to ask them for you.

**Comment by ESadek-MO** [comment]
@bjlittle, @SciTools/peloton were wondering if you'd be happy to make a table of options (including forgoing a CoC) which could be voted on?

**Comment by bjlittle** [comment]
> @bjlittle, @SciTools/peloton were wondering if you'd be happy to make a table of options (including forgoing a CoC) which could be voted on?

Done. See the updated issue description.

How long do you guys want the vote to be open?

**Comment by lbdreyer** [comment]
> How long do you guys want the vote to be open?

Would a month be okay? If that's too long we could go for 2 weeks?
We just need enough time for people to investigate the options if they want to vote

Also, I wonder whether this should have been a discussion rather than an issue?

**Comment by bjlittle** [comment]
> > I need to do more reading later, but I expect this is in the same space:
> > https://fair-software.eu/
> > [![fair-software.eu](https://camo.githubusercontent.com/9a74b21b0c4c076977018603e207add4af89be37d4b549c3f1061ed718771003/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f666169722d2d736f6674776172652e65752d2545322539372538462532302532302545322539372538462532302532302545322539372538462532302532302545322539372538462532302532302545322539372538422d79656c6c6f77)](https://fair-software.eu)
> > I've seen a few of these badges around.
> 
> That's from the [eScience Center](https://www.esciencecenter.nl/) where @bouweandela works (if I'm not mistaken)

If I'm not mistaken, this seems to be more "general guiding principles" (borrowed from [FAIR](https://www.nature.com/articles/sdata201618)) applied to research software development, rather than a CoC, right?
